src/miniinspectorui.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * @license Copyright (c) 2003-2022, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import React, { Component } from 'react';
+
+import Tree from './components/tree/tree';
+import {
+	getEditorModelRanges,
+	getEditorModelMarkers,
+	getEditorModelTreeDefinition
+} from './model/data/utils';
+import {
+	getEditorViewRanges,
+	getEditorViewTreeDefinition
+} from './view/data/utils';
+
+import './ui.css';
+import './miniinspectorui.css';
+
+export default class MiniInspectorUI extends Component {
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			...getTreeDefinitions( this.props.editor )
+		};
+	}
+	render() {
+		return <div className="ck-inspector ck-mini-inspector">
+			<Tree
+				definition={this.state.modelTreeDefinition}
+				textDirection="ltr"
+				onClick={() => {}}
+				showCompactText={false}
+				activeNode={null}
+			/>
+			<Tree
+				definition={this.state.viewTreeDefinition}
+				textDirection="ltr"
+				onClick={() => {}}
+				showCompactText="true"
+				showElementTypes={false}
+				activeNode={null}
+			/>
+		</div>;
+	}
+
+	componentDidMount() {
+		this.props.editor.model.document.on( 'change', () => {
```

Comment:
This bit is slightly inaccurate (I know, this was my idea). I used the `lowest` priority to ensure the view has been rendered on changes in the model but this is not necessarily the truth. In some cases, the view is rendered independently from the model, for instance, when the editor gets focused/blurred/read-only. So, long story short, we need two listeners here: `document#change` and `view#render` with the same callback.

Reply:
Done. I assume we still want to keep priority `lowest` but just for these two events.